### PR TITLE
Port AI static to vis_contents

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
refactor: AI static now uses visual contents and should perform better in theory.
/:cl:

Intended as groundwork for fixing the click issues with #38255.

Old model:
* Every turf which has ever been viewed by an AI has an image
* Obscured turfs get this image added to the `client.images` of AI eyes within its range

New model:
* One effect object and one image with the effect object as its loc
* Obscured turfs have the effect object in their `vis_contents`
* AI eyes have the image in their `client.images`

Remaining issues:

* [x] This [BYOND bug](http://www.byond.com/forum/?post=2378173)
* [ ] Another BYOND bug with no reliable repro where stuff in `vis_contents` appears on the wrong turfs
  * Also applies to gas overlays
  * Fixed by reconnecting
  * Might have something to do with flying around the map very quickly
* [x] Obscured turfs becoming unobscured when `ChangeTurf` clears `vis_contents`
* [x] Use of fractional planes which I'm pretty sure aren't real

### CODE REVIEW

**Comment by optimumtact** [comment]
ME OH MY I ENJOYED THAT YES BOY

**Comment by Cruix** [comment]
I totally misunderstood what you wanted, this might just work.

**Comment by ninjanomnom** [comment]
>Obscured turfs becoming unobscured when ChangeTurf clears vis_contents

Just re-apply it in ChangeTurf, I'd prefer we treat vis_contents basically like overlays for the most part.

**Comment by optimumtact** [comment]
are you going to explain what that bug is or link to it so anyone can tell you? Don't just assume people know what you're referring to

**Comment by Anonmare** [comment]
@optimumtact This one #38595

**Comment by ninjanomnom** [comment]
No it wont

**Comment by SpaceManiac** [comment]
The immediately relevant BYOND bug has been fixed in 512.1428, the occasional no-consistent-repro corruption isn't really a blocker, so this should be ready to go.

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -325,6 +326,14 @@
 
 /turf/proc/visibilityChanged()
 	GLOB.cameranet.updateVisibility(src)
+	// The cameranet usually handles this for us, but if we've just been
+	// recreated we should make sure we have the cameranet vis_contents.
+	var/datum/camerachunk/C = GLOB.cameranet.chunkGenerated(x, y, z)
+	if(C)
+		if(C.obscuredTurfs[src])
+			vis_contents += GLOB.cameranet.vis_contents
```
[the remark this answers is not in the document]
Comment (SpaceManiac) [reply code/game/turfs/turf.dm]: Part of `vis_contents`'s status as a special list is that it can never contain duplicates.

**Review by ninjanomnom — APPROVED** [review]
Seems fine but let's get some testing first

**Comment by Cyberboss** [comment]
Been on for a day

**Comment by optimumtact** [comment]
this was not ready at all http://imgur.com/a/MkS0Lt9 and should be reverted

**Pull request merged**